src/knot_cloud.c

```diff
@@ -549,7 +549,7 @@ int knot_cloud_list_devices(void)
  *
  * Returns: 0 if successful and a KNoT error otherwise.
  */
-int knot_cloud_publish_data(const char *id, uint8_t sensor_id,
```

Comment:
Why change from uint8_t to int? What do you think about including a comment explaining the reason for using int?